python-checks/src/main/java/org/sonar/python/checks/SklearnEstimatorHyperparametersCheck.java

```diff
@@ -0,0 +1,178 @@
+  }
+
+  private static void checkEstimator(SubscriptionContext ctx) {
+    CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+
+    Symbol calleeSymbol = callExpression.calleeSymbol();
+
+    Optional.ofNullable(calleeSymbol)
+      .filter(callee -> callee.is(Kind.CLASS))
+      .map(ClassSymbol.class::cast)
+      .map(ClassSymbol::fullyQualifiedName)
+      .map(ESTIMATORS_AND_PARAMETERS_TO_CHECK::get)
+      .filter(parameters -> !isSetParamsCalled(callExpression))
+      .filter(parameters -> !isPartOfPipelineAndGridSearch(callExpression))
+      .filter(parameters -> isMissingAHyperparameter(callExpression, parameters))
+      .ifPresent(parameters -> ctx.addIssue(callExpression, MESSAGE));
+  }
+
+  private static boolean isMissingAHyperparameter(CallExpression callExpression, List<Param> parametersToCheck) {
+    return parametersToCheck.stream()
+      .map(param -> {
+        if (param.position.isPresent()) {
+          return TreeUtils.nthArgumentOrKeyword(param.position.get(), param.name, callExpression.arguments());
+        }
+        return TreeUtils.argumentByKeyword(param.name, callExpression.arguments());
+      })
+      .anyMatch(Objects::isNull);
+  }
+
+  private static boolean isPartOfPipelineAndGridSearch(CallExpression callExpression) {
+    return getPipelineAssignement(callExpression)
+      .map(SklearnEstimatorHyperparametersCheck::isPipelineUsedInGridSearch)
+      .orElse(false);
+  }
+
+  private static Optional<Name> getPipelineAssignement(CallExpression callExpression) {
+    return Optional.ofNullable(TreeUtils.firstAncestorOfKind(callExpression, CALL_EXPR))
+      .flatMap(TreeUtils.toOptionalInstanceOfMapper(CallExpressionImpl.class))
+      .filter(Objects::nonNull)
+      .filter(callExp -> Optional.ofNullable(callExp.calleeSymbol())
+        .map(Symbol::fullyQualifiedName)
+        .map(fqn -> "sklearn.pipeline.make_pipeline".equals(fqn) || "sklearn.pipeline.Pipeline".equals(fqn))
+        .orElse(false))
+      .flatMap(Expressions::getAssignedName);
+  }
+
+  private static boolean isPipelineUsedInGridSearch(Name pipeline) {
+    return Optional.ofNullable(pipeline.symbol())
+      .map(Symbol::usages)
+      .map(usages -> usages.stream()
+        .map(Usage::tree)
+        .map(Tree::parent)
+        .filter(parent -> parent.is(Tree.Kind.REGULAR_ARGUMENT))
+        .map(RegularArgument.class::cast)
+        .anyMatch(SklearnEstimatorHyperparametersCheck::isArgumentPartOfGridSearch))
+      .orElse(false);
+  }
+
+  private static boolean isArgumentPartOfGridSearch(RegularArgument arg) {
+    return Optional.ofNullable(TreeUtils.firstAncestorOfKind(arg, CALL_EXPR))
+      .flatMap(toOptionalInstanceOfMapper(CallExpressionImpl.class))
+      .map(CallExpressionImpl::calleeSymbol)
+      .map(Symbol::fullyQualifiedName)
+      .map("sklearn.model_selection._search.GridSearchCV"::equals)
```

Comment:
There are 3 other FQNs that we should check to avoid FPs

Reply:
You are talking about the RandomSearch and the HalvingGrid and HalvingRandom search right?

Reply:
Yes, they are the ones